src/classes/DynamicMapOfCountry.cls

```diff
@@ -0,0 +1,161 @@
+                Total_District__c = (Decimal)aggre.get('totalDistrict')));
+        }
+        if(!mapOfStateToBeUpdate.isEmpty()){
+            update mapOfStateToBeUpdate.values();
+        }
+        //Aggregate Qurey------> Total Sate AND District Of Country
+        Map<Id, Country__c> mapOfCountryToBeUpdate = new Map<Id, Country__c>();
+        for (AggregateResult agg : [SELECT Country__c , COUNT(Id)totalStates, SUM(Total_District__c)totalDistrict
+                                    FROM State__c 
+                                    GROUP BY Country__c]){ 
+            mapOfCountryToBeUpdate.put(String.valueOf(agg.get('Country__c')),new Country__c(Id = (Id)agg.get('Country__c'),
+                Total_States__c = (Decimal)agg.get('totalStates'),
+                Total_District__c = (Decimal)agg.get('totalDistrict')));
+        }
+        if(!mapOfCountryToBeUpdate.isEmpty()){
+            update mapOfCountryToBeUpdate.values();   
+        }
+    }
+    /**   Description :   JSON Data
+    *     Created By  :   Sagar Nirwal
+    *     Arguments   :   String
+    *     Return      :   void
+    *     Revision log:   V_1.0 - Created  - Sagar Nirwal - 14/07/2022
+    *                     V_1.1 - Modified - Sagar Nirwal - 14/07/2022 - JSON Data 
+    **/    
+    public static void countryData(String countryName){    
+        // JSON Data
+        List<State__c> stateList = [SELECT Id,country__r.Id,country__r.Country_Code__c,country__r.Name,country__r.Total_States__c,
+                                    country__r.Total_District__c, Name
+                                    FROM State__c];
+        if(!stateList.isEmpty()){
+            Country countryWrapper = new Country();
+            countryWrapper.countryName = stateList[0].country__r.Name;
+            countryWrapper.countryCode = stateList[0].country__r.Country_Code__c;
+            countryWrapper.totalStates = stateList[0].country__r.Total_States__c;
+            countryWrapper.totalDistricts = stateList[0].country__r.Total_District__c;
+            countryWrapper.states = new List<State>();
+            for(State__c st : stateList){
+               State stateWrapper = new State();
+                stateWrapper.stateName = st.Name;
+                stateWrapper.districts = new List<District>();
+                if(countryMap.containsKey(countryName)&& countryMap.get(countryName).containsKey(st.name)){
+                    for(String dist: countryMap.get(countryName).get(st.Name)){
+                        District districtWrapper = new District();
+                        districtWrapper.districtName = dist;
+                        stateWrapper.districts.add(districtWrapper);
+                    }
+                }
+                countryWrapper.states.add(stateWrapper);
+            }
+            JSONData = JSON.serialize(countryWrapper);
+            System.debug('===>>>'+JSONData);   
+        }
+        if(!stateList.isEmpty()){
```

Comment:
You have same if on line 100 no need to add if again just move your code in above if.